src/js/services/ng-score.js

```diff
@@ -0,0 +1,77 @@
+/**
+ * This is the object representation of a score.
+ * It currently contains only the score's summary, but the goal is for it to
+ * fully contain the score from creation to saving, editing and deleting.
+ */
+define('services/ng-score',[
+    'services/ng-services'
+],function(module) {
+    "use strict";
+
+    return module.factory('$score',
+        [function() {
+
+            function Score(entry) {
+
+                //Adding the data from the entry, snitizing it if needed.
+                (function(score, entry) {
+                    let sanitized = Score.sanitize(entry);
+                    for(var key in sanitized) {
+                        score[key] = sanitized[key];
+                    }
+                }) (this, entry);
+
+            }
+
+            Score.prototype.calcFilename = function() {
+                this.file =  [
+                    'score',
+                    this.stageId,
+                    'round' + this.round,
+                    'table' + this.table,
+                    'team' + this.teamNumber,
+                    this.id
+                ].join('_')+'.json';
+                return this.file;
+            };
+
+            Score.compare = function(score1, score2) {
+                return score2.score - score1.score;
+            };
+
+            // These two functions has no reason to be public except for UT.
+
+            // Calculating the unique ID for this sanitized score.
+            // The uid is an 8-hex-digit combination of
+            // The current date and a random number.
+            Score.generateUniqueId = function() {
+                //The max uid in the range
+                let max = 0x100000000;
+                // The numeric form of the uid
+                let num = (Math.floor(Math.random() * max) + Date.now()) % max;
+                // The string uid, by adding the max value and then removing it we make sure the stringification of the number
+                return (num + max).toString(16).slice(1);
+            };
+
+            // This function is meant to make sure the score is set according to
+            // the correct structure in order to save it.
+            // score that doesn't match the fields in this function,
+            // cannot be saved in the scores summery.
+            Score.sanitize = function(entry) {
+                return {
+                    id: entry.id || Score.generateUniqueId(),
```

Comment:
As I wrote above, maybe the server need some sort of validation. Otherwise, we should leave it like this, since the system's data folder is starting off fresh this year (with new file systems on new instances of the program), and that leaves us at a point where we can change the way things work to be more reliable ( i.e., using ids for the scores). In that case, no client will have a score without an id, and so he won't push a score without an id

Reply:
In an ideal world, yes, the scores will always be correct. In the real world, people may be using slightly outdated cached versions in their browser, people making small bugs, angular encouraging mutate-all-state, etc.

So, please add validation in the server, too.

Also solves my personal case, where I constantly use older test data, and would hate to have to recreate it everytime someone changes something like this.

Reply:
Will add validation